### PR TITLE
Fix #409 do not list directory when using s3 as state backend

### LINKED ISSUE #409: directory are listed when we use s3 as a backend for states
**Description**
Sometimes directory are listed when we use s3 as a backend for states.
Reading them produce a "no state" error and stop the scan.

**How to reproduce**
create an empty inside a bucket using aws web console. You may add statefile in it.
Tun scan with this bucket as a backend.

### PR DESCRIPTION
**Pull request by lotoussa**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | yes
| 🚀 New feature?   | no
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #409 
| ❓ Documentation  | no

## Description

- Ignore directory (size <= 0) when listing bucket object
- Add test for empty directory

### CODE REVIEW

File: pkg/iac/terraform/state/enumerator/s3.go [hunk 1]
```diff
@@ -45,7 +46,9 @@ func (s *S3Enumerator) Enumerate() ([]string, error) {
 	}
 	err := s.client.ListObjectsV2Pages(input, func(output *s3.ListObjectsV2Output, lastPage bool) bool {
 		for _, metadata := range output.Contents {
-			keys = append(keys, strings.Join([]string{bucket, *metadata.Key}, "/"))
+			if aws.Int64Value(metadata.Size) > 0 {
```
Comment (eliecharra) [inline pkg/iac/terraform/state/enumerator/s3.go:49]: We should add a test case where this value is `nil`

File: pkg/iac/terraform/state/enumerator/s3_test.go [hunk 2]
```diff
@@ -73,6 +79,48 @@ func TestS3Enumerator_Enumerate(t *testing.T) {
 				"bucket-name/a/nested/prefix/state6",
 			},
 		},
+        {
+            name: "test results are not returned",
```
Comment (eliecharra) [inline pkg/iac/terraform/state/enumerator/s3_test.go:83]: `test that directories objects are filtered`

**Review by eliecharra — CHANGES_REQUESTED** [review]

**Review by eliecharra — APPROVED** [review]
